src/components/form/cancel-button/__spec__.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { elementsTagTest, rootTagTest } from '../../../utils/helpers/tags/tags-specs';
+import { shallow } from 'enzyme';
+import CancelButton from './cancel-button';
+import Button from './../../button';
+
+describe('CancelButton', () => {
+ let wrapper, clickSpy;
+
+ beforeEach(() => {
+   wrapper = shallow(
+     <CancelButton
+       cancelText='This is a cancel button'
+     />
+   );
+ });
+
+  describe('the standard class', () => {
+    it('renders it', () => {
+      let button = wrapper.find('.carbon-form-cancel__button');
+      expect(wrapper.hasClass('carbon-form-cancel')).toBeTruthy();
+    });
+  });
+
+  describe('the save text', () => {
+    describe('when custom saveText is passed in', () => {
+      it('renders it', () => {
+        let button = wrapper.find('.carbon-form-cancel__button');
+        expect(button.prop('children')).toEqual('This is a cancel button');
+      });
+    });
+
+    describe('when no custom text is passed in', () => {
+       beforeEach(() => {
+         wrapper = shallow(<CancelButton />);
+       });
+
+      it('renders the default', () => {
+        let button = wrapper.find('.carbon-form-cancel__button');
+        expect(button.prop('children')).toEqual('Cancel');
+      });
+    });
+  });
+
+  describe('the button classes', () => {
+    describe('when a custom class is passed in', () => {
+      beforeEach(() => {
+        wrapper = shallow(
+         <CancelButton
+           cancelButtonProps={ { className: 'my-custom-button-class' } }
+         />
+       );
+     });
+
+      it('outputs it', () => {
+        expect(wrapper.find('.my-custom-button-class').exists()).toBeTruthy;
+        expect(wrapper.find('.carbon-form-cancel__button')).toBeFalsy;
+      });
+    });
+  });
+
+  describe('clicking cancel', () => {
```

Comment:
This block can probablhy be converted to:

`it('adds the onClick prop to the button')`

Then put the `beforeEach` contents in the `it`